model/mac-command.cc

```diff
@@ -590,7 +590,7 @@ RxParamSetupReq::Serialize(Buffer::Iterator& start) const
     start.WriteU8(GetCIDFromMacCommand(m_commandType));
     // Data serialization
     start.WriteU8((m_rx1DrOffset & 0b111) << 4 | (m_rx2DataRate & 0b1111));
-    uint32_t encodedFrequency = uint32_t(m_frequency / 100);
+    auto encodedFrequency = uint32_t(m_frequency / 100);
```

Comment:
Same as above, if possible use
uint32_t encodedFrequency = m_frequency / 100;

Reply:
To add to @pagmatt , in the event that casting is necessary, use static_casting instead:
static_casting<uint32_t> (m_frequency) /100

Reply:
I agree with @pagmatt that the cast is not necessary, I'll try to fix this everywhere.

One of my proposed changes for the future is in fact to directly represent frequencies in Hz with uint32_t as it happens in [real LoRaWAN implementations](https://github.com/Lora-net/packet_forwarder/blob/d0226eae6e7b6bbaec6117d0d2372bf17819c438/lora_pkt_fwd/src/lora_pkt_fwd.c#L231C1-L234C92) instead of MHz and doubles.